python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -406,10 +406,34 @@ def default_metadata_from_dbt_resource_props(
     dbt_resource_props: Mapping[str, Any],
 ) -> Mapping[str, Any]:
     metadata: Dict[str, Any] = {}
+
+    column_schema = None
     columns = dbt_resource_props.get("columns", {})
     if len(columns) > 0:
-        return dict(
-            TableMetadataSet(
+        column_schema = TableSchema(
+            columns=[
+                TableColumn(
+                    name=column_name,
+                    type=column_info.get("data_type") or "?",
+                    description=column_info.get("description"),
+                )
+                for column_name, column_info in columns.items()
+            ]
+        )
+
+    # all nodes should have these props defined, but just in case
+    relation_identifier = None
+    if (
+        "database" in dbt_resource_props
+        and "schema" in dbt_resource_props
+        and "name" in dbt_resource_props
+    ):
+        relation_identifier = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['name']}"
```

Comment:
We should probably use the `relation_name` property, rather than constructing this by hand.

Otherwise, instead of "name", we should use "alias", which configures the underlying name of the table in the database.
```suggestion
    if (
        "database" in dbt_resource_props
        and "schema" in dbt_resource_props
        and "alias" in dbt_resource_props
    ):
        relation_identifier = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['alias']}"
```

Reply:
Hmm, yeah, the only reason I hesitated to use `relation_name` is that it wasn't specified as required in the dbt schema (while the above keys are) - not sure if in practice that's the case though.

Reply:
We use it when building column lineage. It's pretty safe, as it's actually populated for models, seeds, snapshots, etc.

Reply:
is "{database}.{schema}.{table}" a valid identifier in postgres?

`relation_name` is attractive if it helps abstract over the intricacies of different databases